playground/redirect_route.go

```diff
@@ -0,0 +1,43 @@
+package playground
+
+import (
+	"io"
+
+	"github.com/kkrull/gohttp/http"
+	"github.com/kkrull/gohttp/msg"
+)
+
+func NewRedirectRoute() http.Route {
+	return &RedirectRoute{
+		Resource: &GoBackHomeResource{},
+	}
+}
+
+type RedirectRoute struct {
+	Resource RelocatedResource
+}
+
+func (route *RedirectRoute) Route(requested http.RequestMessage) http.Request {
+	if requested.Path() != "/redirect" {
+		return nil
+	}
+
+	return requested.MakeResourceRequest(route.Resource)
+}
+
+type RelocatedResource interface {
+	Name() string
+}
+
+type GoBackHomeResource struct{}
+
+func (*GoBackHomeResource) Name() string {
+	return "Relocated Resource"
+}
+
+func (*GoBackHomeResource) Get(client io.Writer, req http.RequestMessage) {
+	msg.WriteStatusLine(client, 302, "Found")
```

Comment:
Status codes and messages are also things I would consider wrapping in an enum

Reply:
Agreed.  I modified `msg.WriteStatusLine` and to accept pre-defined `msg.Status` objects.  They're not guarded against mutation or re-assignment, but this seems to follow the pattern in the Go standard libraries nonetheless.